c_src/filter.h

```diff
@@ -328,19 +328,21 @@ class EqOperator<unsigned char*>: public BinaryExpression<bool, unsigned char*>
     virtual ~EqOperator() {};
 
     virtual bool evaluate() const {
+        // [] to represent NULL w/i Riak TS, which does NOT conflict w/ <<"">>.
+        if(right_->size() == 0) {
+            return !left_->has_value();
+        }
```

Comment:
Your understanding is correct, but to ensure it is clearer in the future, I've added comments to the equality comparison for NULL stating that the NULL is always on the right-hand-side.